### PR TITLE
feat: [Issuer Adapter] Support for state value between issuer and adapter redirects

### LINKED ISSUE #104: [Issuer Adapter] support for state value in redirect urls between issuer and issuer adapter
The issuer would need to send the state value in the first redirect and the adapter would need to send back this unmodified value to the issuer once connection with wallet is established.

### PR DESCRIPTION
**Pull request by rolsonquadras**
closes #104 

Signed-off-by: Rolson Quadras <rolson.quadras@securekey.com>

### CODE REVIEW

File: pkg/restapi/issuer/operation/operations_test.go [hunk 1]
```diff
@@ -229,7 +232,34 @@ func TestConnectWallet(t *testing.T) {
...
+		err = c.profileStore.SaveProfile(data)
+		require.NoError(t, err)
+
+		walletConnectHandler := getHandler(t, c, endpoint)
+
+		urlVars[idPathParam] = profileID
+
+		rr := serveHTTPMux(t, walletConnectHandler, walletConnectEndpoint, nil, urlVars)
+
+		fmt.Println(rr.Body.String())
```
Comment (sudeshrshetty) [inline pkg/restapi/issuer/operation/operations_test.go:257]: leftover?

Comment (rolsonquadras) [reply pkg/restapi/issuer/operation/operations_test.go]: Yeah, Thanks. Removed now.

File: pkg/restapi/issuer/operation/operations.go [hunk 2]
```diff
@@ -40,6 +41,8 @@ const (
 	// http params
 	idPathParam     = "id"
 	txnIDQueryParam = "txnID"
+	stateQueryParam = "state"
+	redirectURLFmt  = "%s?state=%s&code=%s"
```
Comment (llorllale) [inline pkg/restapi/issuer/operation/operations.go:45]: this looks too similar to an oauth2 request

Comment (rolsonquadras) [reply pkg/restapi/issuer/operation/operations.go:45]: yes (intentional - used similar names).

Comment (llorllale) [reply pkg/restapi/issuer/operation/operations.go:45]: please use different names in order to avoid confusion

Comment (rolsonquadras) [reply pkg/restapi/issuer/operation/operations.go:45]: Sure, please suggest.

Comment (llorllale) [reply pkg/restapi/issuer/operation/operations.go:45]: why isn't an OTP used? the connectionID would not need to leave the adapter

Comment (rolsonquadras) [reply pkg/restapi/issuer/operation/operations.go:45]: Yeah, I was planning to update it as part of next PR when this OTP is really used. For now, I wanted to define the API.

Comment (llorllale) [reply pkg/restapi/issuer/operation/operations.go:45]: `code` should be removed and instead just use the `state` value to correlate the redirect with the callback.

`connectionID` is too stable and is basically an identifier that should never leave the adapter

Comment (rolsonquadras) [reply pkg/restapi/issuer/operation/operations.go:45]: code is for correlating in adapter, I dont want to use state.

Comment (llorllale) [reply pkg/restapi/issuer/operation/operations.go:45]: connectionID should not be leaked

**Comment by rolsonquadras** [comment]
> leaking connectionID

As I mentioned, I was trying to define the API surface and planning to remove connectionID with some mapping. I'll hardcode some value and add a TODO.

File: pkg/restapi/issuer/operation/operations.go (hunk 2 above)
Comment (rolsonquadras) [reply pkg/restapi/issuer/operation/operations.go:45]: https://github.com/trustbloc/edge-adapter/pull/105/files#r443821604

**Pull request merged**